### PR TITLE
[vcpkg scripts] New: vcpkg_remove_empty_dirs

### PR DESCRIPTION
**Pull request by dg0yt**
Code de-duplication.

### CODE REVIEW

**Comment by dg0yt** [comment]
mongo-c-driver has an installation order problem with the recently added cyrus-sasl: 
~~~
CMake Error at /mnt/vcpkg-ci/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find SASL2 (missing: SASL2_VERSION SASL2_INCLUDE_DIR
  SASL2_LIBRARY) (Required is at least version "2.0")
Call Stack (most recent call first):
  /mnt/vcpkg-ci/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /mnt/vcpkg-ci/installed/x64-linux/share/mongoc-1.0/3rdParty/FindSASL2.cmake:64 (find_package_handle_standard_args)
  /mnt/vss/_work/1/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /mnt/vcpkg-ci/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/share/cmake-3.29/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /mnt/vcpkg-ci/installed/x64-linux/share/mongoc-1.0/mongoc-1.0-config.cmake:46 (find_dependency)
~~~

**Comment by vicroms** [comment]
@AugP @BillyONeal @data-queue @JavierMatosD @ras0219-msft @vicroms 

@ras0219-msft the original argument against this function is that packagers should know ahead of time what directories the package produces and be explicit about deleting them.

We think that if we are providing this feature, then it should be done on all ports by the tool rather than having to explicitly call this function in a portfile. 

We have a handful of options:
1) Status quo, nightmare ports have to deal with code duplication
2) (this PR) extract the common functionality noting that this will always move ports towards successfully building, any port can write this function at the end of their portfile and be more likely to build. 
3) let the executable do this for every portfile, because of (2)

Under current implementation philosophy, empty dirs right are expected to be rare enough that their existence can be used as an error detection mechanism, in which case removing them always removes the possibility of using this as a diagnostic. If they are common enought that manipulation is burdensome then we should go with option 3

**Comment by dg0yt** [comment]
TLDR: This PR is rejected.

**Comment by dg0yt** [comment]
Maybe `vcpkg_cmake_config_fixup` should use this, too, when removing the parent directory?
#41004 isn't only failure to apply fixup but also unexpected removal of lib/cmake/FakeIt when removing the parent of the non-existent lib/cmake/fakeit.
